.github/workflows/static-checks-self-hosted.yaml

```diff
@@ -0,0 +1,130 @@
+on:
+  pull_request_target:
+    branches:
+      - 'main'
+      - 'stable-*'
+    types:
+      - opened
+      - synchronize
+      - reopened
+      - labeled # a workflow runs only when the 'ok-to-test' label is added
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+name: Static checks
+jobs:
+  build-checks:
```

Comment:
You will need to guard this job with `if: ${{ contains(github.event.pull_request.labels.*.name, 'ok-to-test') }}` (e.g. in https://github.com/kata-containers/kata-containers/blob/main/.github/workflows/ci-on-push.yaml#L23) otherwise it will be triggered to any label.

Related with the label, does it make sense to use the same `ok-to-test` label to trigger the static-checks? Because that same label triggers the e2e jobs, all will race for the same bare-metal machines at same time. What's your opinion?

Reply:
I think adding another `label` would make reviewers confused in term of:

- Which label should be added first and under which circumstance (precedence issue)

And once both labels are added, two sets of tests will react to the same set of events, which race for the same resources in the end.

I will keep the config for `ok-to-test`. But I won't make any objection against your suggestion. Thanks.

Reply:
Good point @BbolroC . An extra label means extra complex == more confusion :)

Let's stick with 'ok-to-test` then.